### PR TITLE
Adds deprecation notices for AWSConfig and StorageConfig

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/10599

## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.
- [x] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

**Review by anvddriesch — APPROVED** [review]

**Review by cornelius-keller — APPROVED** [review]

**Comment by tfussell** [comment]
Are we really deprecating StorageConfig? I haven't heard about a plan for its replacement.

**Comment by tfussell** [comment]
Since the CRDs are generated from code now, you'll need to add this notice as a comment on the struct rather than in the CRD YAML. Add here https://github.com/giantswarm/apiextensions/blob/master/pkg/apis/provider/v1alpha1/aws_types.go#L47 and then run `make`. The result should be the same.

File: config/crd/v1/core.giantswarm.io_storageconfigs.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ spec:
   - name: v1alpha1
     schema:
       openAPIV3Schema:
+        description: StorageConfig used to provide storage for Giant Swarm API microservices. Deprecated.
```
Comment (kopiczko) [inline config/crd/v1/core.giantswarm.io_storageconfigs.yaml:25]: What do we use instead?

**Review by marians — APPROVED** [review]

**Comment by marians** [comment]
@JosephSalisbury Please run the `make` command to adapt generated CRD YAML files and then commit the result.

**Comment by marians** [comment]
I'll take over

**Comment by tfussell** [comment]
Why is StorageConfig deprecated?

**Comment by MarcelMue** [comment]
> Why is StorageConfig deprecated?

+1 Two people asked this and then it just got merged?

**Pull request merged**